Update CRD descriptions

dhaiducek opened the pull request:
ref: https://issues.redhat.com/browse/ACM-8992

Along with the description updates, I found some unused structs in the `_type.go` file that I've removed and one that was unused but in the CRD, so I marked it as unused/deprecated.

yiraeChristineKim commented on api/v1/configurationpolicy_types.go:
```diff
@@ -15,31 +15,18 @@ import (
 )
 
-// A custom type is required since there is no way to have a kubebuilder marker
-// apply to the items of a slice.
```
why removed? I expected replacement

yiraeChristineKim replied:
good good

dhaiducek commented:
/hold for reviews

dhaiducek commented on api/v1/configurationpolicy_types.go:
```diff
@@ -15,31 +15,18 @@ import (
-// RemediationAction : enforce or inform
+// RemediationAction specifies the remediation of the policy. The parameter values are enforce and
+// inform.
+//
```
I came up with my own linting rules for this. It separates the kubebuilder tags from the description.

dhaiducek replied:
We can take it out if others don't like it.

yiraeChristineKim replied:
Oh no worry it is good

yiraeChristineKim commented on api/v1/configurationpolicy_types.go:
```diff
@@ -148,61 +119,54 @@ func (e EvaluationInterval) parseInterval(interval string) (time.Duration, error
-	// ObjectDefinition defines required fields for the object
-	// +kubebuilder:pruning:PreserveUnknownFields
-	ObjectDefinition runtime.RawExtension `json:"objectDefinition"`
+	// MustOnlyHave is a ComplianceType to match an object definition exactly with the object.
```
I am sure what mustonlyHave does but the previous "MustOnlyHave is an enforcement state to exclusively include a resource" and this one doesn't match(I feel). exclusive vs exact

dhaiducek replied:
This particular text doesn't wind up in the CRD, so I'm not sure the wording is entirely important here (I thought about removing it... 😬 ). 

Personally I think my wording is more accurate than "exclusively".

yiraeChristineKim replied:
Good, just curious, Can I use `musthaveonly` in case that name or namespace is not specified?

dhaiducek replied:
No, the same rules apply where the config-policy-controller needs to have a name to handle an object. The namespace could be supplied by the `namespaceSelector`, though.

yiraeChristineKim commented on api/v1/configurationpolicy_types.go:
```diff
@@ -211,120 +175,133 @@ const (
+	ObjectDefinition runtime.RawExtension `json:"objectDefinition"`
+
+	// RecordDiff specifies whether (and where) to log the diff between the object on the
+	// cluster and the objectDefinition in the policy. Defaults to "None".
```
Maybe it is better to write the "log" option to this explanation?!!

dhaiducek replied:
Good point--updated!

dhaiducek commented on api/v1beta1/operatorpolicy_types.go:
```diff
@@ -10,100 +10,129 @@ import (
-	// DeleteIfUnused is a RemovalBehavior indicating that the controller may delete
-	// a type only if is not being used by another subscription
+
+	// DeleteIfUnused is a RemovalBehavior indicating that the controller may delete a type only if is
```
There's another line below this one--is that what you're looking for?

dhaiducek replied:
Oh right--good catch!

yiraeChristineKim commented:
Wow a lot of work Great Dale

mprahl commented on api/v1/configurationpolicy_types.go:
```diff
@@ -50,48 +37,29 @@ func (ra RemediationAction) IsEnforce() bool {
+// PruneObjectBehavior is used to remove objects that are managed by the policy upon policy
+// deletion.
+//
+// +kubebuilder:validation:Enum=DeleteAll;DeleteIfCreated;None;
```
```suggestion
// +kubebuilder:validation:Enum=DeleteAll;DeleteIfCreated;None
```

mprahl commented on api/v1/configurationpolicy_types.go:
```diff
@@ -211,120 +175,134 @@ const (
-//+kubebuilder:subresource:status
-//+kubebuilder:printcolumn:name="Compliance state",type="string",JSONPath=".status.compliant"
+	// NamespaceSelector defines the list of namespaces to include/exclude for objects defined in
+	// spec.object-templates. All selector rules are ANDed. If 'include' is not provided but
```
Nitpick: Since there is a hyphen, this is how you reference it with JSON syntax.  :)
```suggestion
	// spec["object-templates"]. All selector rules are ANDed. If 'include' is not provided but
```

mprahl commented on api/v1/configurationpolicy_types.go:
```diff
@@ -211,120 +175,134 @@ const (
+	// have Go templates. For more advanced Go templating such as `range` loops and `if` conditionals, use
+	// 'object-templates-raw'. Only one of 'object-templates' and 'object-templates-raw' may be set in a configuration
+	// policy.
+	// For more on the Go templates, see https://github.com/stolostron/go-template-utils/blob/main/README.md
```
I like this link in upstream but I'm thinking we probably want to avoid it in the ACM version (i.e. Stolostron).

dhaiducek replied:
That's true. For `stolostron`, we could have a doc link instead (though I'm not a huge fan of keeping up with the version there).

mprahl commented on api/v1/configurationpolicy_types.go:
```diff
@@ -341,23 +320,80 @@ func ObjectResourceFromObj(obj client.Object) ObjectResource {
+	Object ObjectResource `json:"object,omitempty"`
+
+	// Compliant is whether the related object is compliant with the configuration policy's
+	// definition(s).
```
```suggestion
	// definition.
```

mprahl commented on api/v1/configurationpolicy_types.go:
```diff
@@ -341,23 +320,80 @@ func ObjectResourceFromObj(obj client.Object) ObjectResource {
+// contains, in whole or in part, an object definition to compare with objects on the cluster. If
+// the configuration policy's definition doesn't match the objects on the cluster, the configuration
+// policy will report a non-compliant status. Furthermore, if the RemediationAction is set to
+// "enforce" and the name of the object is available, the configuration policy controller will patch
```
```suggestion
// "enforce" and the name of the object is available, the configuration policy controller will create or update
```

mprahl reviewed (dismissed):
Great work!

dockerymick commented on deploy/crds/kustomize_configurationpolicy/policy.open-cluster-management.io_configurationpolicies.yaml:
```diff
@@ -218,46 +238,62 @@ spec:
             properties:
               compliancyDetails:
+                description: |-
+                  CompliancyDetails is a list of statuses matching 1-1 with each of the items in the
+                  object-templates array.
```
```suggestion
                  CompliancyDetails is a list of statuses matching one-to-one with each of the items in the
                  'object-templates' array.
```

dockerymick commented on deploy/crds/kustomize_configurationpolicy/policy.open-cluster-management.io_configurationpolicies.yaml:
```diff
@@ -218,46 +238,62 @@ spec:
                 items:
-                  description: TemplateStatus hold the status result
+                  description: |-
+                    TemplateStatus reports the compliance details resulting from the definitions in an
+                    object-template.
```
```suggestion
                    TemplateStatus reports the compliance details from the definitions in an
                    'object-template'.
```

dockerymick commented on deploy/crds/kustomize_configurationpolicy/policy.open-cluster-management.io_configurationpolicies.yaml:
```diff
@@ -218,46 +238,62 @@ spec:
                   properties:
                     Compliant:
-                      description: ComplianceState shows the state of enforcement
+                      description: ComplianceState reports the observed status resulting
+                        from the policy's definition.
```
```suggestion
                      description: ComplianceState reports the observed status
                        from the definition of the policy.
```

dockerymick commented on deploy/crds/kustomize_configurationpolicy/policy.open-cluster-management.io_configurationpolicies.yaml:
```diff
@@ -218,46 +238,62 @@ spec:
                           type: boolean
                       type: object
                     conditions:
+                      description: Conditions contains the details from the latest
+                        evaluation of the object-template.
```
```suggestion
                      description: Conditions contains the details from the latest
                        evaluation of the 'object-template'.
```

dockerymick commented on deploy/crds/kustomize_configurationpolicy/policy.open-cluster-management.io_configurationpolicies.yaml:
```diff
@@ -218,46 +238,62 @@ spec:
-                        description: Condition is the base struct for representing
-                          resource conditions
+                        description: Condition contains the details of an evaluation
+                          of an object-template.
```
same suggestion

dockerymick commented on deploy/crds/kustomize_configurationpolicy/policy.open-cluster-management.io_configurationpolicies.yaml:
```diff
@@ -218,46 +238,62 @@ spec:
                             type: string
                           reason:
-                            description: The reason for the condition's last transition.
+                            description: Reason is a brief reason for the condition.
```
```suggestion
                            description: Reason is a brief summary for the condition.
```

dockerymick commented on deploy/crds/kustomize_configurationpolicy/policy.open-cluster-management.io_configurationpolicies.yaml:
```diff
@@ -218,46 +238,62 @@ spec:
                             type: string
                           type:
-                            description: Type of condition, e.g Complete or Failed.
+                            description: Type is the type of condition, either "violation"
+                              or "notification".
```
```suggestion
                            description: Type is the type of condition. The supported values are "violation"
                              or "notification".
```

dockerymick commented on deploy/crds/kustomize_configurationpolicy/policy.open-cluster-management.io_configurationpolicies.yaml:
```diff
@@ -266,64 +302,77 @@ spec:
                 type: array
               compliant:
-                description: ComplianceState shows the state of enforcement
+                description: ComplianceState reports the observed status resulting
+                  from the policy's definition.
```
```suggestion
                description: ComplianceState reports the observed status
                  from the definition of the policy.
```

dockerymick commented on deploy/crds/kustomize_configurationpolicy/policy.open-cluster-management.io_configurationpolicies.yaml:
```diff
@@ -266,64 +302,77 @@ spec:
               relatedObjects:
-                description: List of resources processed by the policy
+                description: |-
+                  RelatedObjects is a list of objects processed by the configuration policy due to its
+                  object-templates.
```
single quotes for object-template

dockerymick commented on deploy/crds/kustomize_configurationpolicy/policy.open-cluster-management.io_configurationpolicies.yaml:
```diff
@@ -266,64 +302,77 @@ spec:
                   properties:
                     compliant:
+                      description: |-
+                        Compliant is whether the related object is compliant with the configuration policy's
+                        definition.
```
```suggestion
                        Compliant is whether the related object is compliant with the definition of the configuration policy.
```

dockerymick reviewed (changes requested):
still going through the rest of the YAMLs

dhaiducek commented on deploy/crds/kustomize_configurationpolicy/policy.open-cluster-management.io_configurationpolicies.yaml:
```diff
@@ -159,14 +173,16 @@ spec:
                       description: |-
-                        RecordDiff specifies whether (and where) to log the diff between the object on the
-                        cluster and the objectDefinition in the policy. Defaults to "None".
+                        RecordDiff specifies whether (and where) to log the diff between the object on the cluster and
+                        the objectDefinition in the policy. Options are "Log" to log the diff in the
+                        config-policy-controller pod and "None" to not log the diff. The default value is "None".
```
I've used "difference" here, the reason being that it could be logged when the remediation action is `inform` and the object is different but wouldn't be updated.

dhaiducek commented on deploy/crds/kustomize_configurationpolicy/policy.open-cluster-management.io_configurationpolicies.yaml:
```diff
@@ -159,14 +173,16 @@ spec:
-                      description: ObjectDefinition defines required fields for the
-                        object
+                      description: |-
+                        ObjectDefinition defines required fields, in whole or in part, to be compared with objects on
+                        the cluster.
```
The goal is to state that the `ObjectDefinition` doesn't have to be fully defined. A partial object definition can be supplied and the controller will work with that to patch the object on the managed cluster.

dhaiducek replied:
That said, it's probably find to just leave it out. It seems "required fields" might imply that.

dhaiducek commented on deploy/crds/kustomize_configurationpolicy/policy.open-cluster-management.io_configurationpolicies.yaml:
```diff
@@ -42,52 +47,56 @@ spec:
           spec:
-            description: ConfigurationPolicySpec defines the desired state of ConfigurationPolicy
+            description: |-
+              ConfigurationPolicySpec defines the desired configuration of objects on the cluster along with
+              how the configuration policy should handle when the cluster doesn't match the configuration
+              policy.
```
The user isn't doing the handling here--it's the controller, so should I say that instead? Something like:
```
// ConfigurationPolicySpec defines the desired configuration of objects on the cluster, along with
// how the controller should handle when the cluster doesn't match the configuration policy.
```

dhaiducek commented:
@dockerymick I think I addressed the comments. Please take another look when you have a moment. (I'll take another look too when I'm not going cross-eyed looking at it 🤣 )

dockerymick commented on deploy/crds/kustomize_operatorpolicy/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -142,13 +174,22 @@ spec:
           status:
-            description: OperatorPolicyStatus defines the observed state of OperatorPolicy
+            description: |-
+              OperatorPolicyStatus reports the observed state of the operators resulting from the
+              specifications given in the operator policy.
```
```suggestion
              OperatorPolicyStatus is the observed state of the operators that are from the
              specifications given in the operator policy.
```

dockerymick commented on deploy/crds/kustomize_operatorpolicy/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -142,13 +174,22 @@ spec:
                 type: string
               conditions:
-                description: Historic details on the condition of the policy
+                description: Conditions reports historic details on the condition
+                  of the operator policy.
```
```suggestion
                description: Conditions include historic details on the condition
                  of the operator policy.
```

dockerymick commented on deploy/crds/kustomize_operatorpolicy/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -221,52 +262,57 @@ spec:
+                    by the configuration policy.
                   properties:
                     compliant:
+                      description: |-
+                        Compliant is whether the related object is compliant with the definition of the configuration
+                        policy.
```
Trying to get away from using the parameter name in the description. in understand that when policies are compliant, the object meets the configuration requirements
```suggestion
                      description: |-
                        Compliant represents whether the related object meets the defined conditions of the configuration
                        policy.
```

dockerymick reviewed (changes requested):
Next up, third YAML review. I noticed that the single quotes are used inconsistently. Even though the backticks are not supported in YAML, I would be consistent and use backticks when we reference parameter values.

dhaiducek commented on deploy/crds/kustomize_operatorpolicy/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -37,11 +40,14 @@ spec:
             type: object
           spec:
-            description: OperatorPolicySpec defines the desired state of OperatorPolicy
+            description: OperatorPolicySpec defines the desired state of a particular
+              operator on the cluster.
```
Correct--it's user-defined.

dockerymick commented on deploy/crds/kustomize_operatorpolicy/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -124,15 +159,16 @@ spec:
                 description: |-
-                  Include the namespace, and any `spec` fields for the Subscription.
-                  For more info, see `kubectl explain subscription.spec` or
+                  Subscription specifies which operator Subscription to inspect. Include the namespace, and any
+                  `spec` fields for the Subscription. For more info, see `kubectl explain
+                  subscriptions.operators.coreos.com.spec` or view
```
```suggestion
                  Subscription specifies which operator `Subscription` resource to inspect. Include the namespace, and any
                  `spec` fields for the Subscription. For more info, see `kubectl explain
                  subscriptions.operators.coreos.com.spec` or view.
```

dockerymick commented on deploy/crds/kustomize_operatorpolicy/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -124,15 +159,16 @@ spec:
-                  Versions is a list of nonempty strings that specifies which installed versions are compliant when
-                  in 'inform' mode, and which installPlans are approved when in 'enforce' mode
+                  Versions is a list of non-empty strings that specifies which installed versions are compliant
+                  when in 'inform' mode and which InstallPlans are approved when in 'enforce' mode.
```
```suggestion
                  when in `inform` mode and which `InstallPlans` are approved when in `enforce` mode.
```

dockerymick commented on deploy/crds/kustomize_operatorpolicy/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -142,13 +178,22 @@ spec:
           status:
-            description: OperatorPolicyStatus defines the observed state of OperatorPolicy
+            description: |-
+              OperatorPolicyStatus is the observed state of the operators resulting from the
+              specifications given in the operator policy.
```
```suggestion
              OperatorPolicyStatus is the observed state of the operators from the
              specifications given in the operator policy.
```

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -19,7 +19,10 @@ spec:
       openAPIV3Schema:
-        description: OperatorPolicy is the Schema for the operatorpolicies API
+        description: |-
+          OperatorPolicy is the Schema for the operatorpolicies API. You can use the operator policy to
+          manage operators by providing automation for their management and reporting on the status across
+          the various operator objects.
```
```suggestion
          OperatorPolicy is the schema for the operatorpolicies API. You can use the operator policy to
          manage operators by providing automation for their management and reporting on the status across
          the various operator objects.
```

dockerymick replied:
```suggestion
           OperatorPolicy is the schema for the operatorpolicies API. You can use the operator policy to
          manage operators by providing automation for their management and reporting on the status across
          the various operator objects.
```

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -39,24 +42,29 @@ spec:
+                description: |-
+                  ComplianceType describes how objects on the cluster should be compared with the object definition
+                  of the configuration policy. The supported options are "MustHave", "MustOnlyHave", or
+                  "MustNotHave".
```
Similar suggestion about quotation mark consistency. I recommend that we use the backtick

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -39,24 +42,29 @@ spec:
-                  Include the name, namespace, and any `spec` fields for the OperatorGroup.
-                  For more info, see `kubectl explain operatorgroup.spec` or
-                  https://olm.operatorframework.io/docs/concepts/crds/operatorgroup/
+                  OperatorGroup specifies which OperatorGroup to inspect. Include the name, namespace, and any
+                  `spec` fields for the OperatorGroup. For more info, see `kubectl explain operatorgroups.spec`
+                  or view https://olm.operatorframework.io/docs/concepts/crds/operatorgroup/
```
```suggestion
                  OperatorGroup specifies which `OperatorGroup` resource to inspect. Include the name, namespace, and any
                  `spec` fields for the OperatorGroup. For more info, see `kubectl explain operatorgroups.spec`
                  or view https://olm.operatorframework.io/docs/concepts/crds/operatorgroup/.
```

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -39,24 +42,29 @@ spec:
-                description: 'RemediationAction : enforce or inform'
+                description: |-
+                  RemediationAction is the remediation of the policy. The parameter values are "enforce" and
+                  "inform".
```
Use backtick mark

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -66,47 +74,74 @@ spec:
-                  RemovalBehavior defines what resources will be removed by enforced mustnothave policies.
-                  When in inform mode, any resources that would be deleted if the policy was enforced will
-                  be causes for NonCompliance, but resources that would be kept will be considered Compliant.
+                  RemovalBehavior defines what resources need to be removed when enforcing `mustnothave`
+                  policies. When in inform mode, any resources that would be deleted if the policy was enforced
+                  will be causes for NonCompliance, but resources that would be kept will be considered
+                  Compliant.
```
```suggestion
                  Use RemovalBehavior to define what resources need to be removed when enforcing `mustnothave`
                  policies. When in `inform` mode, any resources that are deleted if the policy is set to `enforce`
                  makes the policy non-compliant and resources that are kept are considered
                  compliant.
```

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -66,47 +74,74 @@ spec:
-                    - Delete
+                    description: |-
+                      Use clusterServiceVersions to specify whether to delete the ClusterServiceVersion. The default
+                      value is "Delete".
```
```suggestion
                      value is `Delete`.
```

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -66,47 +74,74 @@ spec:
-                    - Keep
-                    - Delete
+                      Use the customResourceDefinitions parameter to specify whether to delete any
+                      CustomResourceDefinitions associated with the operator. The default value is "Keep", because
```
suggestion about consistency

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -66,47 +74,74 @@ spec:
-                    enum:
-                    - Keep
-                    - DeleteIfUnused
+                      Use the operatorGroups parameter to specify whether to delete the OperatorGroup. The default
+                      value is "DeleteIfUnused", which will only delete the OperatorGroup if there is not another
+                      resource using it.
```
```suggestion
                      Use the operatorGroups parameter to specify whether to delete the `OperatorGroup` resource. The default
                      value is `DeleteIfUnused`, which deletes only the OperatorGroup if there is not another
                      resource using it.
```

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -66,47 +74,74 @@ spec:
-                    - Delete
+                    description: |-
+                      Use the subscriptions parameter to specify whether to delete the Subscription. The default
+                      value is "Delete".
```
Quotation consistency

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -66,47 +74,74 @@ spec:
                 type: object
               severity:
-                description: 'Severity : low, medium, high, or critical'
+                description: |-
+                  Severity is a user-defined severity for when an object is noncompliant with this configuration
+                  policy. The supported options are "low", "medium", "high", and "critical".
```
Quotation mark consistency
```suggestion
                description: |-
                  Severity is a user-defined severity for when an object is non-compliant with this configuration
                  policy. The supported options are "low", "medium", "high", and "critical".
```

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -119,15 +154,16 @@ spec:
                 description: |-
-                  Include the namespace, and any `spec` fields for the Subscription.
-                  For more info, see `kubectl explain subscription.spec` or
+                  Subscription specifies which operator Subscription to inspect. Include the namespace, and any
+                  `spec` fields for the Subscription. For more info, see `kubectl explain
+                  subscriptions.operators.coreos.com.spec` or view
```
```suggestion
                  Subscription specifies which operator `Subscription` resource to inspect. Include the namespace, and any
                  `spec` fields for the Subscription. For more info, see `kubectl explain
                  subscriptions.operators.coreos.com.spec` or view
```

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -119,15 +154,16 @@ spec:
+                  Subscription specifies which operator Subscription to inspect. Include the namespace, and any
+                  `spec` fields for the Subscription. For more info, see `kubectl explain
+                  subscriptions.operators.coreos.com.spec` or view
                   https://olm.operatorframework.io/docs/concepts/crds/subscription/
```
```suggestion
                  https://olm.operatorframework.io/docs/concepts/crds/subscription/.
```

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -119,15 +154,16 @@ spec:
-                  Versions is a list of nonempty strings that specifies which installed versions are compliant when
-                  in 'inform' mode, and which installPlans are approved when in 'enforce' mode
+                  Versions is a list of non-empty strings that specifies which installed versions are compliant
+                  when in 'inform' mode and which InstallPlans are approved when in 'enforce' mode.
```
```suggestion
                  when in `inform` mode and which InstallPlans are approved when in `enforce` mode.
```

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -137,13 +173,22 @@ spec:
             type: object
           status:
-            description: OperatorPolicyStatus defines the observed state of OperatorPolicy
+            description: |-
+              OperatorPolicyStatus is the observed state of the operators resulting from the
+              specifications given in the operator policy.
```
```suggestion
            description: |-
              OperatorPolicyStatus is the observed state of the operators from the
              specifications given in the operator policy.
```

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -216,52 +261,56 @@ spec:
                 items:
-                  description: RelatedObject is the list of objects matched by this
-                    Policy resource.
+                  description: RelatedObject contains the details of an object matched
```
```suggestion
                  description: RelatedObject contain the details of an object matched
```

dhaiducek replied:
`RelatedObject` is singular--should this still be "RelatedObject contains" rather than "RelatedObject contain"?

dockerymick commented on deploy/crds/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -216,52 +261,56 @@ spec:
                     object:
-                      description: ObjectResource is an object identified by the policy
-                        as a resource that needs to be validated.
+                      description: ObjectResource contains the identifying fields
```
```suggestion
                      description: ObjectResource contain the identifying fields
```

dockerymick reviewed (changes requested):
One more policy to go: open-cluster-management.io_configurationpolicies.yaml

dockerymick commented:
I thought I didn't review `policy.open-cluster-management.io_configurationpolicies.yaml`, but i did. i'm not sure why there is a conflict though

dhaiducek commented on deploy/crds/kustomize_operatorpolicy/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -37,11 +40,14 @@ spec:
-                  have a given resource
+                description: |-
+                  ComplianceType describes how objects on the cluster should be compared with the object definition
+                  of the configuration policy. The supported options are "MustHave", "MustOnlyHave", or
```
I'm good with backticks, especially if that's the standard in the `rhacm-docs` repo.

dhaiducek commented:
> I thought I didn't review `policy.open-cluster-management.io_configurationpolicies.yaml`, but i did. i'm not sure why there is a conflict though

Sorry, I'm not following--what conflict are you referring to?
(I did get sidetracked and hadn't gotten to all of the comments yet.)

dhaiducek commented on deploy/crds/kustomize_operatorpolicy/policy.open-cluster-management.io_operatorpolicies.yaml:
```diff
@@ -71,47 +79,74 @@ spec:
               severity:
-                description: 'Severity : low, medium, high, or critical'
+                description: |-
+                  Severity is a user-defined severity for when an object is noncompliant with this configuration
```
It should stay as "noncompliant": https://www.ibm.com/docs/en/ibm-style?topic=grammar-prefixes

dhaiducek commented on api/v1/configurationpolicy_types.go:
```diff
@@ -149,62 +121,89 @@ func (e EvaluationInterval) parseInterval(interval string) (time.Duration, error
-	// RecordDiff specifies whether (and where) to log the diff between the object on the
-	// cluster and the objectDefinition in the policy. Defaults to "None" when the object kind is
-	// ConfigMap, OAuthAccessToken, OAuthAuthorizeTokens, Route, or Secret. Defaults to "InStatus" otherwise.
+	// RecordDiff specifies whether and where to log the difference between the object on the cluster
+	// and the `objectDefinition` in the policy. The supported options are `InStatus` to record the
+	// difference in the policy status field, `Log` to log the difference in the
+	// config-policy-controller pod, and `None` to not log the diff. The default value is `None` for
+	// object kinds that include sensitive data including `ConfigMap`, `OAuthAccessToken`,
+	// `OAuthAuthorizeTokens`, `Route`, and `Secret`. For all other kinds, the default value is
+	// `InStatus`.
```
FYI @mprahl I updated the description for `recordDiff`.

mprahl replied:
Updated here:
https://github.com/open-cluster-management-io/config-policy-controller/pull/248

Thanks for the suggestion!

dhaiducek commented:
/unhold
